app/javascript/components/admin/UserModal.js

```diff
@@ -109,18 +122,62 @@ class UserModal extends React.Component {
               />
             </Form.Group>
             {this.props.type === 'edit' && (
+              <Form.Row className="pt-4">
+                <Form.Group as={Col} md={8}>
+                  <Form.Label>System Access</Form.Label>
+                  <Form.Check
+                    id="system-access-input"
+                    name="system-access"
+                    type="switch"
+                    checked={this.state.isLocked}
+                    label={this.state.isLocked ? 'Locked' : 'Unlocked'}
+                    onChange={this.handleLockedSystemAccessChange}
+                  />
+                </Form.Group>
+                <Form.Group as={Col}>
+                  {this.state.lockReason === 'Auto-locked by the System' && this.state.isLocked && (
+                    <Form.Text as={Alert} id="system-access-auto-lock-reason" variant="light" className="admin-lock-alert">
+                      <i className="fas fa-exclamation-circle"></i>
+                      <span> {this.state.autoLockReason}</span>
+                    </Form.Text>
+                  )}
+                </Form.Group>
+              </Form.Row>
+            )}
+            {this.props.type === 'edit' && this.state.isLocked && (
```

Comment:
Is there a reason there are two different inputs used here instead of using the `isDisabled` flag to disable it when `this.state.isLocked` is true?

Reply:
I was incorrectly trying to use `disabled` instead of `isDisabled`, which wasn't working (of course) so I used two inputs. I've made this update in [be724b2f805fa4843309cff5302a8a597faefd26](https://github.com/SaraAlert/SaraAlert/pull/1180/commits/be724b2f805fa4843309cff5302a8a597faefd26)

Thank you!

Reply:
looks good!